### PR TITLE
Alerting: Fix contact point testing with secure settings

### PR DESCRIPTION
**Pull request by JacobsonMT**
Fix error when attempting to test a contact point with existing, saved secure settings.

Removes double encryption of secure settings during contact point testing and code duplication that helped cause the drift between alertmanager and test endpoint. Also adds integration tests to cover the regression.

Note: `provisioningStore` is created to prevent a cycle and the unnecessary dependency.

### CODE REVIEW

File: pkg/services/ngalert/notifier/alertmanager_config.go
Comment (JacobsonMT) [inline pkg/services/ngalert/notifier/alertmanager_config.go:189]: Necessary to resolve an import cycle (also removes `github.com/grafana/grafana/pkg/services/ngalert/provisioning` dependency from `ngalert/notifier`)

File: pkg/services/ngalert/notifier/crypto.go
Comment (JacobsonMT) [inline pkg/services/ngalert/notifier/crypto.go:64]: Copied as-is from `definitions`

File: pkg/services/ngalert/notifier/crypto.go
Comment (JacobsonMT) [inline pkg/services/ngalert/notifier/crypto.go:85]: Copied as-is from `definitions`

File: pkg/tests/api/alerting/api_notification_channel_test.go
Comment (JacobsonMT) [inline pkg/tests/api/alerting/api_notification_channel_test.go:242]: Regression test for the issue

File: pkg/services/ngalert/api/api_alertmanager.go
Comment (JacobsonMT) [inline pkg/services/ngalert/api/api_alertmanager.go:304]: This is the bugfix. It's now in-sync with `ApplyAlertmanagerConfiguration`.

**Review by JohnnyQQQQ — APPROVED** [review]
LGTM, but someone else is the right person to verify that. We should get a second review from @yuri-tceretian or @grobinson-grafana, that are more involved in this part of the code.

File: pkg/services/ngalert/notifier/crypto.go
Comment (yuri-tceretian) [inline pkg/services/ngalert/notifier/crypto.go:57]: I am not sure that this is crypto's responsibility.

**Review by yuri-tceretian — APPROVED** [review]
LGTM. I am a bit hesitant about the responsibilities of the new method ProcessReceivers.

File: pkg/services/ngalert/notifier/crypto.go
Comment (JacobsonMT) [reply pkg/services/ngalert/notifier/crypto.go:57]: That's fair, I can move it out of `ProcessReceivers` then give it a better name like `ProcessSecureSettings`. I doubt this is even necessary for testing receivers... if not, I can keep it as a function in `alertmanager_config`.

Comment (JacobsonMT) [reply pkg/services/ngalert/notifier/crypto.go]: c29df0960482ef4f2e4cf65701ad00c6d43cb97b

**Comment by ElijahLynn** [comment]
Thank you! This fixed an issue we were having with a 400 Bad Request error when hitting the test button on a contact point!
image

**Pull request merged**
